azurerm/data_source_netapp_volume.go

```diff
@@ -0,0 +1,141 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	aznetapp "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/netapp"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmNetAppVolume() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmNetAppVolumeRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: aznetapp.ValidateNetAppPoolName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			"location": azure.SchemaLocationForDataSource(),
+
+			"account_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: aznetapp.ValidateNetAppAccountName,
+			},
+
+			"pool_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: aznetapp.ValidateNetAppPoolName,
+			},
+
+			"creation_token": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"service_level": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"subnet_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"usage_threshold": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"export_policy_rule": {
```

Comment:
I think we can remove this field - since I don't think it's likely to be used by users?

Reply:
I don't think so. Now both powershell and azure portal for netapp volume have opened the entry to let user edit export policy. And we cannot ensure the users won't use this. If they want to use, I submit another PR for this? Why don't we leave it as optional field? And this field should be optional. See the doc https://docs.microsoft.com/en-us/azure/azure-netapp-files/azure-netapp-files-configure-export-policy.

Reply:
whilst users may configure this on the Resource, the Data Source is Read Only and I can't envisage a scenario where users are likely to use this in the Data Source (since this needs to be a Set too) - for now let's remove it and we can add it back in if users request it

Reply:
removed